File: day 73/classwork/cw3.js
// 3) შექმენით ნებისმიერ ფუნქცია, რომელსაც გამოიძახებთ ყოველ 3 წამში ერთხელ, შემდეგ ეს ისტრუქცია დააკომენტარეთ და ფუნქცია გამოძახეთ ერთხელ ინსტრუქციის წაკითხვიდან 4 წამის შემდეგ, კომენტარებით ახსენით როგორ მუშაობს setInterval და setTimeout ფუნქციები

let myFunction = function() {
    console.log("Hello, world!");
}
setInterval(myFunction, 3000); 

setTimeout(myFunction, 4000); 

File: Day 054/classwork/cw3.py
# 3) შექმენით decorator ფუნქცია, რომელიც თავისთავად შექმნის wrapper ფუნქციას და დააბრუენბს მას. აიღეთ მარტივი greet ფუნქცია რომელიც აბრუნებს "Hello World"-ს და გამოიძახეთ მიანიჭეთ მას decorator. საბოლოოდ გამოიძახეთ greet ფუნქცია


def my_decorator(func):
    def wrapper():
        print("Something is happening before the function is called.")
        result = func()
        print("Something is happening after the function is called.")
        return result
    return wrapper
@my_decorator
def greet():
    return "Hello World" 